lib/cli/watch-run.js

```diff
@@ -347,12 +335,18 @@ const eraseLine = () => {
 
 /**
  * Blast all of the watched files out of `require.cache`
- * @param {FSWatcher} watcher - chokidar FSWatcher
  * @ignore
  * @private
  */
-const blastCache = watcher => {
-  const files = getWatchedFiles(watcher);
+const blastCache = () => {
+  const files = Object.keys(require.cache)
+    // Avoid deleting mocha binary
+    .filter(
+      file =>
+        !file.includes('mocha/bin/mocha.js') &&
```

Comment:
Also doing some tests, to avoid extra invalidations, this should be `node_modules/mocha/`. Confirmed that that single filtering works with Mocha tests, standard mocha dependency/package, and a custom repository & branch path (at least under PNPM).